crates/compiler/can/src/constraint.rs

```diff
@@ -770,6 +840,14 @@ pub enum Constraint {
         Index<PatternCategory>,
         Region,
     ),
+    /// Check call fx against enclosing function fx
+    FxCall(Index<FxCallConstraint>),
+    /// Require idents to be accurately suffixed
+    FxSuffix(Index<FxSuffixConstraint>),
+    /// Set an fx var as pure if flex (no effectful functions were called)
+    FlexToPure(Variable),
```

Comment:
Is this necessary? Would it be sufficient to add an Eq constraint that the fx var on a function is Pure or Impure depending on the suffix? With an appropriate context.

Reply:
Do you mean `FlexToPure` or `FxSuffix` should use `Eq`? 

The former I think wouldn't work because the logic is different than the normal unification of fx vars. However, `FxSuffix` might work as `Eq` with a custom `Reason`.

Reply:
@ayazhafiz let me know which one you mean when you have a chance

Reply:
We decided to merge this, but I'm happy to clean this up in a follow up PR!